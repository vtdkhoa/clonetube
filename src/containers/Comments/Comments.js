import React from 'react'
import CommentsHeader from './CommentsHeader/CommentsHeader'
import AddComment from './AddComment/AddComment'
import Comment from './Comment/Comment'
import PropTypes from 'prop-types'

const Comments = ({ comments, amountComments }) => {
  if (!comments) {
    return <div/>
  }

  const showComments = getComments(comments)
  return (
    <div className="comments" style={{ marginTop: 0 }}>
      <CommentsHeader amountComments={amountComments}/>
      <AddComment/>
      {showComments}
    </div>
  )
}

const getComments = comments => {
  return comments.map(comment =>
    <Comment
      comment={comment}
      key={comment.id}
    />
  )
}

Comments.propTypes = {
  comments: PropTypes.array,
  amountComments: PropTypes.any
}

export default Comments