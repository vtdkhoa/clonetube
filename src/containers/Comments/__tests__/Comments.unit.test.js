import React from 'react'
import Comments from '../Comments'
import { shallow } from 'enzyme'

describe('Comments', () => {
  test('renders without props', () => {
    const wrapper = shallow(<Comments/>)
    expect(wrapper).toMatchSnapshot()
  })

  test('renders with amountComments', () => {
    const wrapper = shallow(
      <Comments amountComments={1000000}/>
    )
    expect(wrapper).toMatchSnapshot()
  })
})