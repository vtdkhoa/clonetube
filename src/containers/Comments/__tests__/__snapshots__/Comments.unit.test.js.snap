// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments renders with amountComments 1`] = `
<div>
  <CommentsHeader
    amountComments={1000000}
  />
  <AddComment />
  <Comment />
  <Comment />
  <Comment />
</div>
`;

exports[`Comments renders without props 1`] = `
<div>
  <CommentsHeader />
  <AddComment />
  <Comment />
  <Comment />
  <Comment />
</div>
`;
